src/args.rs

```diff
@@ -14,6 +14,10 @@ pub struct Args {
     /// Parameters for the model, each in the form `key=value`
     #[clap(short, long)]
     pub parameters: Vec<String>,
+
+    // Model deviation tolerance
+    #[clap[short, long, default_value = "0.0"]]
+    pub tolerance: f64,
```

Comment:
I don't like how an invalid value is used as the default. I realize this is handled correctly below, but it's still weird, and also potentially misleading, as this information is used for the auto-generated command-line documentation.

How about:
```suggestion
    #[clap[short, long]]
    pub tolerance: Option<f64>,
```
I *think* that should work. Then handle the `Option` accordingly. If the user specifies an invalid value (zero or less), that can result in an error message.

I believe you should be able to call `anyhow::bail!("error message: {}", some_value)` in `main` to produce a nice error message. If that doesn't work, a panic will do for now.

Reply:
Thanks for the feedback!